pkg/pmem-csi-operator/controller/deployment/deployment_controller.go

```diff
@@ -120,6 +124,7 @@ func NewReconcileDeployment(client client.Client, opts pmemcontroller.Controller
 		namespace:      opts.Namespace,
 		containerImage: opts.DriverImage,
 		deployments:    map[string]*pmemcsiv1alpha1.Deployment{},
+		reconcileHooks: []ReconcileHook{},
```

Comment:
You can drop this initialization.

Reply:
With the latest change, the `reconcileHook`  is a `map`, that requires initialization.